product/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import json
+import bcrypt
+import jwt
 
-# Create your views here.
+from django.http              import HttpResponse, JsonResponse
+from django.views             import View
+
+from datetime                 import date, datetime
+from .models                  import Category, Product, Reward, Collection, LikeUser, ProductContent
+
+
+class CategoryView(View):
+    def get(self, request):
+        try:
+            product = Product.objects.all()
+            product_list = [{
+                'category_image'   : product.category_set.all()[0].image,
+                'title'            : product.title,
+                'goal_amout'       : product.goal_amount,
+                'toal_amount'      : product.total_amount,
+                'achieved_rate'    : product.achieved_rate,
+                'total_supporters' : product.total_supporters,
+                'closing_date'     : product.closing_date,
+                'thumbnail'        : product.thumbnail_url,
+                'category'         : product.category_set.all()[0].name,
+                'id'               : product.id,
+                'maker_info_name'  : product.maker_info.name
+                } for product in product]
+            return JsonResponse({"MESSAGE" : "SUCCESS", "data" : {'product'  : product_list}}, status=200)
```

Comment:
- list comprehension 적용 완벽합니다 👍
- 가독성을 위해 빈 줄 추가해주시면 더욱 더 깔끔해지겠네요!
- `Product.objects.all()` 의 결과물은 QuerySet list 이므로 변수명도 복수형으로 지어주세요!
- `all()[0]` 을 `first()` 라는 메소드로 대체할 수도 있으니 참고바랍니다 :)
```suggestion
            product = Product.objects.all()
            
            product_list = [{
                'category_image'   : product.category_set.all()[0].image,
                'title'            : product.title,
                'goal_amout'       : product.goal_amount,
                'toal_amount'      : product.total_amount,
                'achieved_rate'    : product.achieved_rate,
                'total_supporters' : product.total_supporters,
                'closing_date'     : product.closing_date,
                'thumbnail'        : product.thumbnail_url,
                'category'         : product.category_set.all()[0].name,
                'id'               : product.id,
                'maker_info_name'  : product.maker_info.name
                } for product in product]
                
            return JsonResponse({"MESSAGE" : "SUCCESS", "data" : {'product'  : product_list}}, status=200)
```

Reply:
오호 꿀팁 감사합니다.!!! 또 하나 배워 갑니다.!!